File: BackEnd/src/domain/session.ts
import { PlanItem } from './plan-item';

export class Session {
  sessionId: string;
  teacherId: string;
  subjectId: string;
  day: string;
  items: PlanItem[];
  maxDuration: number; //later we get it from the teacher
  usedDuration: number;
  reviewBufferMinutes: number;
  slotNumber: number;
  sessionDate: Date;
  sessionWeekNo: number;
}

File: Frontend/GP/Models/Session.cs
using System.ComponentModel.DataAnnotations;

namespace GP.Models
{
    public class Session : BaseEntity
    {
        [Key]
        public int SessionId { get; set; }
        public int TeacherId { get; set; }
        public Teacher Teacher {  get; set; }

        public int  SubjectId { get; set; }
        public Subject Subject { get; set; }
        public string Day { get; set; }
        public List<PlanItem> Items { get; set; }
        public int MaxDuration { get; set; }
        public int UsedDuration { get; set; }
        public int ReviewBufferMinutes { get; set; }
        public int SlotNumber { get; set; }
        public DateTime SessionDate { get; set; }
        public int SessionWeekNo { get; set; }
    }
}
